helix-term/src/ui/editor.rs

```diff
@@ -430,7 +434,7 @@ impl EditorView {
                     viewport.x + (i * tab_width as u16) - offset.col as u16,
                     viewport.y + line,
                     &indent_guide_char,
-                    indent_style,
+                    text_style.patch(indent_guide_style),
```

Comment:
Is it a good idea to patch `ui.virtual.indent-guide` if it exists? I think it should only patch `ui.virtual.whitespace` as a fallback rather than patching `ui.virtual.indent-guide` if it exists. I am still undecided since users may put `ui.virtual.indent-guide` without specifying a background.

Also, rather than doing this in a loop, I think it might be better if we patch it directly to `indent_guide_style` rather than doing this in the inner function.

Reply:
The reason I'm patching it with `text_style` is that if users only specify the foreground, it should use the background from `text_style`. If you look at line 457, all styles are patched from `text_style`
```rust
let style = spans
    .iter()
    .fold(text_style, |acc, span| acc.patch(theme.highlight(span.0)));
```
You're right about doing it outside of the loop, will fix.